strawberryfields/engine.py

```diff
@@ -280,15 +278,12 @@ def _broadcast_nones(val, dim):
                 p = p.compile(target, **compile_options)
             p.lock()
 
-            if self.REMOTE:
-                self.samples = self._run_program(p, **kwargs)
-            else:
-                self._run_program(p, **kwargs)
-                shots = kwargs.get("shots", 1)
-                self.samples = [
-                    _broadcast_nones(p.reg_refs[k].val, shots) for k in sorted(p.reg_refs)
-                ]
-                self.run_progs.append(p)
+            self._run_program(p, **kwargs)
+            shots = kwargs.get("shots", 1)
+            self.samples = [
+                _broadcast_nones(p.reg_refs[k].val, shots) for k in sorted(p.reg_refs)
```

Comment:
@josh146 @thisac `_broadcast_nones` is candidate for removal in future PR

Reply:
I'm not sure, but I guess so if the `None`s should be removed from samples somehow. `_broadcast_nones` merely takes a single `None` in a mode and reshapes it into a list of `None`s with length `shots`, right?

Reply:
What should the placeholder be instead of `None` when there are no results in that case? @josh146 can probably give a better answer to this.

Reply:
@co9olguy, how come? Currently `_broadcast_nones` is used a lot in the local simulators; when a mode is not terminated in measurement, the same result is currently `None`.